AK/HashTable.h

```diff
@@ -262,22 +346,39 @@ class HashTable {
         auto& bucket = lookup_for_writing(value);
         new (bucket.slot()) T(move(value));
         bucket.used = true;
+
+        if constexpr (IsOrdered) {
+            if (!m_housekeeping.head) [[unlikely]] {
+                m_housekeeping.head = &bucket;
+            } else {
+                bucket.previous = m_housekeeping.tail;
+                m_housekeeping.tail->next = &bucket;
+            }
+            m_housekeeping.tail = &bucket;
+        }
     }
 
     void rehash(size_t new_capacity)
```

Comment:
this whole method is currently broken, the next and previous pointers are not handled (they are invalid after this) and the head and tail fields are not updated either

Reply:
wdym

Reply:
the pointers inside the bucket point to freed memory once this method finishes.

Reply:
isn't bucket a reference though?
and how would you fix that then

Reply:
Oh you mean the rehash, I am working on fixing it, but that isnt really working